src/main/scala/firrtl/AddDescriptionNodes.scala

```diff
@@ -89,44 +143,97 @@ class AddDescriptionNodes extends Transform with PreservesAll[Transform] {
 
   override val dependents = Seq.empty
 
-  def onStmt(compMap: Map[String, Seq[String]])(stmt: Statement): Statement = {
-    stmt.map(onStmt(compMap)) match {
-      case d: IsDeclaration if compMap.contains(d.name) =>
-        DescribedStmt(DocString(StringLit.unescape(compMap(d.name).mkString("\n\n"))), d)
-      case other => other
+  def onStmt(compMap: Map[String, Seq[Description]])(stmt: Statement): Statement = {
+    val s = stmt.map(onStmt(compMap))
+    val sname = s match {
+      case d: IsDeclaration => Some(d.name)
+      case _ => None
+    }
```

Comment:
Minor style nit but this could just be
```suggestion
    s.collect { case d: IsDeclaration => d.name }
```